ddtrace/_trace/_span_link.py

```diff
@@ -70,8 +73,8 @@ def name(self, value):
         self.attributes["link.name"] = value
 
     @property
-    def kind(self):
-        return self.attributes["link.kind"]
+    def kind(self) -> Optional[Any]:
```

Comment:
```suggestion
    def kind(self) -> Optional[str]:
```
Is it not always a string? Maybe out of scope.

Reply:
it technically *should* be a string. we can mark it as such on the setter, too? but if we're not going to throw an exception when someone says `link.kind = {"something": "funky"}` i'd prefer to stick with `Any`?

Reply:
ok, changed this and the setter to mark them as expecting an `str`